File: 프로그래머스0/18.조건 문자열.js
/*
문자열에 따라 다음과 같이 두 수의 크기를 비교하려고 합니다.
두 수가 n과 m이라면
">", "=" : n >= m
"<", "=" : n <= m
">", "!" : n > m
"<", "!" : n < m
*/

function solution(ineq, eq, n, m) {
    var answer = 0;
    
    if (ineq === '>' && eq === '='){
        answer = n >= m ? 1 : 0;
    } else if (ineq === '<' && eq === '='){
        answer = n <= m ? 1 : 0;
    } else if (ineq === '>' && eq === '!'){
        answer = n > m ? 1 : 0;
    } else if (ineq === '<' && eq === '!'){
        answer = n < m ? 1 : 0;
    } 
    
    return answer;
}

File: 프로그래머스 코테(java)/프로그래머스0(1~30번)/18.조건 문자열.java
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/181934?language=java

class Solution {
        public int solution(String ineq, String eq, int n, int m) {
                if (ineq.equals(">")) {
                        if (eq.equals("=")) {
                                return n >= m ? 1 : 0;
                        } else if (eq.equals("!")) {
                                return n > m ? 1 : 0;
                        }
                } else if (ineq.equals("<")) {
                        if (eq.equals("=")) {
                                return n <= m ? 1 : 0;
                        } else if (eq.equals("!")) {
                                return n < m ? 1 : 0;
                        }
                }

                return 0;
        }
}